Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -1900,6 +1901,69 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncConcat, (JSGlobalObject* globalObject, Ca
     return JSValue::encode(result);
 }
 
+JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncFill, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto thisValue = callFrame->thisValue().toThis(globalObject, ECMAMode::strict());
+    RETURN_IF_EXCEPTION(scope, { });
+    if (thisValue.isUndefinedOrNull())
+        return throwVMTypeError(globalObject, scope, "Array.prototype.fill requires that |this| not be null or undefined"_s);
+    auto* thisObject = thisValue.toObject(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    uint64_t length = toLength(globalObject, thisObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    int64_t relativeStart = callFrame->argument(1).toIntegerOrInfinity(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+    int64_t k = 0;
+    if (relativeStart < 0) {
+        k = length + relativeStart;
+        if (k < 0)
+            k = 0;
+    } else {
+        k = relativeStart;
+        ASSERT(k >= 0);
+        if (static_cast<uint64_t>(k) > length)
+            k = length;
+    }
+    int64_t relativeEnd = length;
+    JSValue end = callFrame->argument(2);
+    if (!end.isUndefined()) {
+        relativeEnd = end.toIntegerOrInfinity(globalObject);
+        RETURN_IF_EXCEPTION(scope, { });
+    }
+    int64_t final = 0;
+    if (relativeEnd < 0) {
+        final = length + relativeEnd;
+        if (final < 0)
+            final = 0;
+    } else {
+        final = relativeEnd;
+        ASSERT(final >= 0);
+        if (static_cast<uint64_t>(final) > length)
+            final = length;
+    }
+
+    if (k > final)
+        return JSValue::encode(thisObject);
+
+    if (isJSArray(thisValue)) {
+        auto* array = jsCast<JSArray*>(thisValue);
+        if (array->fastFill(vm, k, final, callFrame->argument(0)))
+            return JSValue::encode(array);
+    }
+
+    for (; k < final; k++) {
+        thisObject->putByIndexInline(globalObject, static_cast<uint64_t>(k), callFrame->argument(0), false);
```

Comment:
Instead of calling `callFrame->argument(0)` repeatedly, we should get it once and use a variable.
Also, why is it `false`?